### PR TITLE
Cropcert entities rework

### PR DESCRIPTION
**Pull request by arunasd463**

### CODE REVIEW

File: src/cropcert/entities/api/CooperativePersonApi.java [hunk 1]
```diff
@@ -48,12 +47,29 @@ public CooperativePersonApi(CooperativePersonService farmerService) {
...
 	@ApiOperation(value = "Get co-operative person by id", response = CooperativePerson.class)
 	public Response find(@Context HttpServletRequest request, @PathParam("id") Long id) {
-		CooperativePerson ccPerson = coPersonService.findById(id);
+		CooperativePerson ccPerson = coPersonService.findByUserId(id);
 		if (ccPerson == null)
 			return Response.status(Status.NO_CONTENT).build();
 		return Response.status(Status.CREATED).entity(ccPerson).build();
 	}
 
+	@Path("cocode/{cocode}")
```
Comment (prakhar-s) [inline src/cropcert/entities/api/CooperativePersonApi.java:56]: put `cccode` in ApiConstants.java class and use it from there.

File: src/cropcert/entities/api/CooperativePersonApi.java [hunk 2]
```diff
@@ -80,7 +96,7 @@ public Response save(@Context HttpServletRequest request, String jsonString) {
 		try {
 			coPerson = coPersonService.save(jsonString);
 			return Response.status(Status.CREATED).entity(coPerson).build();
-		} catch (IOException | JSONException e) {
```
Comment (prakhar-s) [inline src/cropcert/entities/api/CooperativePersonApi.java:83]: Can we verify once if this removal of Exception is required?

File: src/cropcert/entities/api/FactoryPersonApi.java [hunk 3]
```diff
@@ -81,10 +84,10 @@ public Response save(@Context HttpServletRequest request, String jsonString) {
 		try {
 			factoryPerson = factoryPersonService.save(jsonString);
 			return Response.status(Status.CREATED).entity(factoryPerson).build();
-		} catch (IOException | JSONException e) {
```
Comment (prakhar-s) [inline src/cropcert/entities/api/FactoryPersonApi.java:84]: can we verify if this removal of exception necessary?

File: src/cropcert/entities/api/FarmerApi.java [hunk 4]
```diff
@@ -133,7 +132,7 @@ public Response save(@Context HttpServletRequest request, String jsonString) {
 		try {
 			farmer = farmerService.save(jsonString);
 			return Response.status(Status.CREATED).entity(farmer).build();
-		} catch (IOException | JSONException e) {
```
Comment (prakhar-s) [inline src/cropcert/entities/api/FarmerApi.java:136]: verify if this is required

File: src/cropcert/entities/api/ICSManagerApi.java [hunk 5]
```diff
@@ -64,10 +67,10 @@ public Response save(@Context HttpServletRequest request, String jsonString) {
 		try {
 			icsManager = icsManagerService.save(jsonString);
 			return Response.status(Status.CREATED).entity(icsManager).build();
-		} catch (IOException | JSONException e) {
```
Comment (prakhar-s) [inline src/cropcert/entities/api/ICSManagerApi.java:67]: verify if this is required

File: src/cropcert/entities/util/AppUtil.java [hunk 6]
```diff
@@ -1,18 +1,18 @@
...
 import java.util.Arrays;
-import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.EnumMap;
 
 public class AppUtil {
 
-	public static final Map<MODULE, List<String>> ALLOWED_CONTENT_TYPES = new HashMap<>();
+	protected static final Map<MODULE, List<String>> ALLOWED_CONTENT_TYPES = new EnumMap<>(MODULE.class);
```
[the remark this answers is not in the document]
Comment (arunasd463) [reply src/cropcert/entities/util/AppUtil.java:10]: reverted to original state

**Review by prakhar-s — APPROVED** [review]

**Pull request merged**